scripts/dev/check_for_bom_in_idfs.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+
+import codecs
+import io
+import json
+import os
+
+
+def usage():
+    print("""This script verifies the IDFs don\'t have byte order marks.""")
+
+
+current_script_dir = os.path.dirname(os.path.realpath(__file__))
+test_files_dir = os.path.join(current_script_dir, '..', '..', 'testfiles')
+
+for root, dirs, files in os.walk(test_files_dir):
+    for sfile in files:
+        if sfile.endswith('.idf') or sfile.endswith('.imf'):
+            if root == test_files_dir:
+                relative_path = sfile
+            else:
+                folder = os.path.basename(os.path.normpath(root))
+                relative_path = os.path.join(folder, sfile)
+            abs_path = os.path.join(test_files_dir, relative_path)
+            with io.open(abs_path, 'rb') as fd:
+                for i, line in enumerate(fd):
+                    if codecs.BOM_UTF8 in line:
```

Comment:
Check to see if the UTF8 byte order mark is in the bytes of the line and if so, throw a warning